lambda-buffers-codegen/src/LambdaBuffers/Codegen/Typescript/Print.hs

```diff
@@ -0,0 +1,252 @@
+                            let getUncommonSuffix :: Eq a => [a] -> [a] -> ([a], [a])
+                                getUncommonSuffix (l : ls) (r : rs)
+                                  | l == r = getUncommonSuffix ls rs
+                                  | otherwise = (l : ls, r : rs)
+                                getUncommonSuffix l r = (l, r)
+
+                                (selfUncommonSuffix, mnUncommonSuffix) = getUncommonSuffix (selfModName ^. #parts) (mn' ^. #parts)
+                             in "./"
+                                  <>
+                                  -- Move back to the first uncommon
+                                  -- directory, then put the uncommon suffix.
+                                  pretty
+                                    ( Text.intercalate "/" $
+                                        drop 1 (map (const "..") selfUncommonSuffix)
+                                          ++ map (view #name) mnUncommonSuffix
+                                    )
+                                  <> lbTsExt
+                        )
+                    Just pkgName ->
+                      -- Print something like
+                      -- @
+                      -- pkg-name/LambdaBuffers/path/to/module/name.mjs
+                      -- @
+                      -- Why do we include @LambdaBuffers@ there? See
+                      -- "LambdaBuffers.Codegen.Typescript.Syntax.filepathFromModuleName"
+                      pretty (Ts.pkgNameToText pkgName)
+                        <> "/"
+                        <> "LambdaBuffers"
+                        <> "/"
+                        <> PC.withInfoLess mn (\mn' -> pretty $ Text.intercalate "/" (mn' ^.. #parts . Lens.traversed . #name))
+                        <> lbTsExt
+                , printModName' mn
+                )
+            )
+          <$> toList groupedLbImports
+
+      groupedTsImports =
+        -- TODO(jaredponn): resolve this awkward maybe situation for the
+        -- PackageName. We really should never have the Nothing case...
+        Set.fromList [(pkg, mn) | (Just pkg, mn, _tn) <- toList tsTyImports]
+          `Set.union` Set.fromList [(pkg, mn) | (pkg, mn, _) <- toList classImps]
+          `Set.union` Set.fromList [(pkg, mn) | (Just (pkg, mn), _) <- toList valImps]
+
+      tsImportDocs = importQualified . ((\(Ts.MkPackageName pkg) -> pretty pkg) *** (\(Ts.MkModuleName mn) -> pretty mn)) <$> toList groupedTsImports
+
+      importsDoc = vsep $ lbImportDocs ++ tsImportDocs
+   in importsDoc
+  where
+    importQualified :: (Doc ann, Doc ann) -> Doc ann
+    importQualified (pkg, mn) = "import" <+> "*" <+> "as" <+> mn <+> "from" <+> squotes pkg
+
+{- | `collectPackageDeps lbTyImports hsTyImports classImps ruleImps valImps` collects all the package dependencies.
+ Note that LB `lbTyImports` and `ruleImps` are wired by the user (as the user decides on the package name for their schemass).
+-}
```

Comment:
This is supposed to be where the lbTyImports and ruleImports are collected with the help of your `packages` configuration no? How does that work?

Reply:
This looks like a mistake on my part -- good catch with this! I updated it to s.t. it'll actually include the packages from lbTyImports + ruleImps https://github.com/mlabs-haskell/lambda-buffers/pull/161/commits/835b1e03661d924eaa589dc138c4982c12adea0d